### DISCUSSION TITLE
Editor of Edited Volume

### DISCUSSION THREAD

**Discussion by kyledenlinger** [knowledgefutures/pubpub#1942] [Support] [answered]
Hi folks. I'm trying to treat a collection as an edited volume: a single editor for the whole book/collection, and each Pub (chapter) has a distinct author. I can't seem to find anything about this in any documentation. Citations do generate as though they're part of an edited volume (In _Book Title_) but the editor doesn't ever appear in the correct place. 

To wit, what I currently get is:

Author, Billy and Melody Editor. 2022. “Chapter Title.” In _Book Title._ Library Publishing. https://urlpubpubdotorg/pub/id.

What I'd like to get is:

Author, Billy. “Chapter Title.” In _Book Title,_ edited by Melody Editor. Library Publishing, 2022. https://urlpubpubdotorg/pub/id.

(Note that the publication date is also in the wrong place in Chicago).

I've tried:
- adding the editor to the Attribution line in the Collection settings, assigned various roles
- adding the editor as a contributor at the Pub level, assigned various roles
- toggling editor name display on/off at both the Collection and Pub level

Is this possible? Am I missing something?

**Comment by gabestein** [comment, marked as answer]
Hello — no, you're not missing anything, and my apologies for the lack of documentation. At the moment, we don't have support for the editor role, which is a big oversight in our attribution model. We had a discussion last week about some potential ways to fix it, and are hoping to have something next week or the week after. 

In the immediate term, we will most likely update the collection-level attribution to add an option to mark a contributor as an editor (similar to the current list on byline option). Do you think that would work for your use-case?

Mid-term, there's a lot of work we want to do on our contributor model including multiple affiliations linked to ROR IDs, better naming support, etc., but that's a bit of a larger project.

**Reply by kyledenlinger** [reply]
> In the immediate term, we will most likely update the collection-level attribution to add an option to mark a contributor as an editor (similar to the current list on byline option). Do you think that would work for your use-case?

Yes, absolutely. It might be nice if "List on Byline" enabled a dropdown of contributor types, default set to author.

**Reply by gabestein** [reply]
Hey, just letting you know that we've effectively done as described above. At both the pub and collection level, we use roles to determine if someone should be in the citation as an author, editor, translator, or illustrator (roughly, the four roles supported in bibtex/csl/crossref). If someone has list in the byline checked at either the collection of Pub level, they will also appear on the byline (but if their role is, e.g., editor, they will be an editor for the purposes of the citation).

**Reply by kyledenlinger** [reply]
Thanks, Gabe! I noticed that earlier this week. Great change!

Regarding editors/translators/illustrators appearing on the Pub byline, any chance their role could be laid out to differentiate them from authors? "_By Aaron Adams and Bernice Browning_" feels different than "_By Aaron Adams, Edited by Bernice Browning_"

**Reply by gabestein** [reply]
Oh, that's a nice suggestion. I wanna check with some other folks who are using the feature — I know in some cases folks want collection editors to be listed after "by", but I like the idea that if you have one of those specific roles we differentiate.

**Reply by writingpoli** [reply]
We'd love to be able to list the author as separate from the illustrator or artist on the byline.

**Comment by katycroffbell** [comment]
Good morning! I have a similar question as Kyle, but can't figure out how to get the output I want. 

I'd like to separate an author of an individual pub within a volume from the overall list of authors of the same volume. 

Example, here I've set myself & MQ to be authors of the whole volume, and SP to be an editor. 

![image](https://user-images.githubusercontent.com/92737253/176680222-2d4e7ab0-e6a5-4610-b28c-a978f497e55c.png)

But when I go to a piece on which I should be the sole author, everyone shows up:
![image](https://user-images.githubusercontent.com/92737253/176680301-8775ab12-1227-4f54-b61f-ea6a0cc5e8f9.png)

And the citation is kinda right but not quite:

![image](https://user-images.githubusercontent.com/92737253/176680410-c0bcff7f-9ff1-4984-b57b-c56817055660.png)

It should be Bell, KC. (date). Executive Summary. In: Deep-Sea Capacity etc etc., Bell, MQ, SP. 

When I unclick "List on Collection Byline" at the highest setting level, all of the volume authors are removed from the pub, and only SP who is labeled as Editor is included in the citation

![image](https://user-images.githubusercontent.com/92737253/176681067-cb6d2c86-fc2c-4202-af06-1482788f3a1a.png)


Is it possible to separate authors of a pub from authors of a volume/issue?  Or should I treat the high-level authors as editors instead so i have authors of pubs and editors of the volume?

**Reply by gabestein** [reply]
If I understand correctly, you want to be both the editor and the author on a pub. To do that, you'd add yourself to the collection with an editor role, but uncheck byline. Then you'd list yourself as an author on the pub and check byline on the pub itself.

We need to amend the collection byline text to say something else, but it's hard to know exactly what. What that checkbox does is puts every contributor in the byline of the collection AND all pubs within the collection AND in the citation as an author or, if they have a role that CSL supports, as that role. If you don't click that box, only people with non-author roles we know about will be listed in the citation. Perhaps we need one checkbox for citation, one for byline?

**Reply by gabestein** [reply]
Sorry, I think I missed something, which is that you also have a collection-level author that you want to show up in the citation, but not as the main author of the pub in the citation or on the byline. Most citation styles don't really support this type of contributor, either in guidance or in the CSL implementation we use, so I would recommend listing them as an editor at the collection level and un-checking byline if you want them to appear on the citation but not as an author (though some, like APA, don't even list editors on most content types). You could add an additional, second role for them as well that we would at least show in PubPub and deposit to Crossref if you plan to mint a DOI, but the first role would need to be an editor one (Editor or Writing — Review & Editing).

**Reply by katycroffbell** [reply]
Got it, I'll just use collection-level editors instead of authors. thanks!

**Comment by katycroffbell** [comment]
Also, is it possible to add the citation of a pub to the first page of the pdf download? It seems like you have all the pieces in place and it would be easy to implement.

**Reply by gabestein** [reply]
A lot of folks use PubPub for things that are not papers to be cited, so we've shied away from that in the past. If you mint a DOI, it will go on the PDF. One thing we're considering is enabling a few different PDF options for folks, some that look more academic, some that are less, etc.

**Reply by katycroffbell** [reply]
Really? I'm looking at another pub that has a DOI and it's not on the pdf.

**Reply by gabestein** [reply]
Should be. Can you send me the example? May have been from an old pdf template that needs to be regenerated.

**Reply by katycroffbell** [reply]
yep! https://openocean.pubpub.org/pub/here-be-dragons-report/release/2

**Reply by katycroffbell** [reply]
Maybe it needs to be re-released?

**Reply by gabestein** [reply]
That works, but I also just deleted all the cached PDFs in your community so they'll be automatically regenerated next time someone requests them.

**Reply by katycroffbell** [reply]
Just downloaded one and it worked! DOI is on it. Thanks!

But, I do still think that auto-adding the reference to the pdf would be a good idea.
